src/metabase/serialization/dump.clj

```diff
@@ -0,0 +1,294 @@
+(ns metabase.serialization.dump
+  "Serialize a Matabase instance into a directory structure of YAMLs."
+  (:require [clojure.java.io :as io]
+            [clojure.string :as str]
+            [clojure.walk :as walk]
+            [metabase.automagic-dashboards.filters :refer [field-reference?]]
+            [metabase.models
+             [card :refer [Card]]
+             [collection :refer [Collection]]
+             [dashboard :refer [Dashboard]]
+             [dashboard-card :refer [DashboardCard]]
+             [dashboard-card-series :refer [DashboardCardSeries]]
+             [database :refer [Database]]
+             [dependency :refer [Dependency]]
+             [dimension :refer [Dimension]]
+             [field :refer [Field]]
+             [field-values :refer [FieldValues]]
+             [metric :refer [Metric]]
+             [pulse :refer [Pulse]]
+             [pulse-card :refer [PulseCard]]
+             [pulse-channel :refer [PulseChannel]]
+             [segment :refer [Segment]]
+             [setting :as setting]
+             [table :refer [Table]]]
+            [metabase.query-processor.util :as qp.util]
+            [metabase.util :as u]
+            [metabase.util.schema :as su]
+            [schema.core :as s]
+            [toucan.db :as db]
+            [yaml.core :as yaml]))
+
+(defmulti
+  ^{:doc      "Get the logical path for entity `entity`.
+
+               The idea is to replace all IDs with these human readable paths which are also
+               instance-independent, making deserialization eaiser."
+    :private  true
+    :arglists '([entity])}
```

Comment:
It must have been in Clojure 1.9 they fixed docstrings on `defmulti`. Now you don't need to stick them in metadata anymore to get them to show up as `:doc`

Also just like `defn` you can pass a meta attributes map between the docstring and dispatch-function so you can use that for `:arglists`. Check out the new methods in `metabase.driver` for example.